.github/workflows/build.yaml

```diff
@@ -0,0 +1,43 @@
+name: Main build
+
+on:
+    workflow_dispatch:
+    pull_request:
+        branches: [main]
+
+jobs:
+    build-gradle:
+        name: build and Test Gradle 
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Code
+              uses: actions/checkout@v4
+            
+            - name: Setup JDK 
+              uses: actions/setup-java@v4
+              with:
+                java-version: '11'
+                distribution: 'temurin'
+            
+            - name: Setup Gradle
+              uses: gradle/actions/setup-gradle@v3
+              with:
+                gradle-version: 6.9.2
+  
+            - name: Test and Build Gradle
+              env:
+                GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                ORG_GRADLE_PROJECT_signingKeyId:  ${{ secrets.GPG_KEYID }}
+                ORG_GRADLE_PROJECT_signingKey: ${{ secrets.GPG_SECRET }}
+                ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.GPG_PASSPHRASE }}
+              run: |
+                gradle check publish --scan \
+                -PsourceMaven=https://repo.maven.apache.org/maven2/  \
+                -PcentralMaven=https://repo.maven.apache.org/maven2/ \
+                -PtargetMaven=file:${{github.workspace}}/repo  \

```

Comment:
So I believe you want your target maven to be the GH Maven registry where the artefacts will deploy to - so https://maven.pkg.github.com/galasa-dev/gradle/ (replacing galasa-dev with your user for testing of course).

The reason in the Tekton builds the target maven is `/gradle/repo` is because we then take the contents of that directory and deploy that to our custom maven registry at https://development.galasa.dev/main/maven-repo/gradle - see the Dockerfile where we do that: https://github.com/galasa-dev/automation/blob/main/dockerfiles/gradle/gradle-dockerfile